### PR TITLE
Added support for drpg slash commands

### PR DESCRIPTION
**Pull request by RagingLink**
This is very much a very quick solution and hasn't been tested for padv (even though it *should* work fine).

This also should not be deployed when drpg has not fully migrated to slash commands, as the code for the timers/health monitor have been written with slash commands in mind.

### CODE REVIEW

**Review by Ciborn — COMMENTED** [review]
For the sake of professionalism, I'll formalize my thoughts here.

Aside from some debatable code style changes you have made, everything looks good! Depending on how long the DiscordRPG beta will be going, it may be necessary to support both "raw input" and slash commands.
Nothing is bad about your own code style as it seems both yours and mine are being used in other projects, but I made some changes to it for the sake of consistency.

As you suggested, this pull request as is will be merged once the DiscordRPG is done. Thank you very much for your contribution!

**Comment by RagingLink** [comment]
aight bet bro

File: src/events/message.ts [hunk 1]
```diff
@@ -21,28 +21,31 @@ export default async (client: AldebaranClient, message: Message) => {
 		: "";
 
 	const author = await client.customUsers.fetch(message.author.id);
+	const interactionUser = message.interaction !== null ? 
+		await client.customUsers.fetch(message.interaction.user.id) : 
+		undefined;
```
Comment (Ciborn) [inline src/events/message.ts]: Considering current Aldebaran's code style, the operators should be placed after the line break.
```suggestion
	const interactionUser = message.interaction !== null
		? await client.customUsers.fetch(message.interaction.user.id)
		: undefined;
```

File: src/structures/contexts/DiscordMessageContext.ts [hunk 2]
```diff
@@ -8,16 +8,15 @@ import Command from "../../groups/Command.js";
 
 export default class DiscordMessageContext extends MessageContext {
 	private _splitArgs: string[];
-	private message: Message;
 	public command?: Command;
-	public author: User;
-	public guild?: Guild;
 
-	constructor(client: Client, message: Message, author: User, guild?: Guild) {
+	constructor(client: Client,
```
Comment (Ciborn) [inline src/structures/contexts/DiscordMessageContext.ts]: `client`'s access specifier should be explicited, and the parameter should be placed after a line break.
```suggestion
	constructor(
		public client: Client, 
```

File: src/structures/contexts/DiscordSlashMessageContext.ts [hunk 3]
```diff
@@ -9,21 +9,15 @@ import MessageContext from "./MessageContext.js";
 type M = Promise<Message<boolean>>;
 
 export default class DiscordSlashMessageContext extends MessageContext {
-	private interaction: CommandInteraction;
-	public author: User;
 	public command: Command;
-	public guild?: Guild;
 
 	constructor(
 		client: Client,
```
Comment (Ciborn) [inline src/structures/contexts/DiscordSlashMessageContext.ts]: `client`'s access specifier should be explicited.
```suggestion
		public client: Client,
```

**Review by Ciborn — CHANGES_REQUESTED** [review]
i hope this works

**Comment by RagingLink** [comment]
I will be looking into adding support for text drpg so this commit doesn't need to be timed with slash drpg.

**Pull request closed without merging**